### PR TITLE
Add L5 Weapon/Armor upgrades

### PR DESCRIPTION
**Pull request by Ziktofel**
Adds L5 Weapon/Armor upgrades

Makes Nova affected by Weapon/Armor upgrades

Sets Ultra Capacitor/Vanadium research times to same value as the basic upgrades

### CODE REVIEW

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml [hunk 1]
```diff
@@ -2308,36 +2308,76 @@
...
             <Resource index="Minerals" value="250"/>
             <Resource index="Vespene" value="250"/>
             <Button DefaultButtonFace="AP_TerranInfantryArmorVanadiumPlatingLevel3" State="Restricted" Requirements="AP_LearnTerranInfantryArmorVanadiumPlating3"/>
         </InfoArray>
+        <InfoArray index="Research17" Time="250" Upgrade="AP_TerranInfantryWeaponsLevel4">
+            <Resource index="Minerals" value="325"/>
+            <Resource index="Vespene" value="325"/>
+            <Button DefaultButtonFace="AP_TerranInfantryWeaponsLevel4" State="Restricted" Requirements="AP_LearnTerranInfantryWeapon4"/>
+        </InfoArray>
+        <InfoArray index="Research18" Time="280" Upgrade="AP_TerranInfantryWeaponsLevel5">
```
Comment (MatthewMarinets) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml:2346]: I'd caution against setting the time to research too high, I'd even recommend capping the times at level 3 times. 160+190+220+250+280=1100 or about 18 minutes 20 seconds. That's longer than the vast majority of missions.

Checking co-op game data, Tychus upgrades are 120+140+160+180+200 = 800. That affects fewer units, so maybe we don't go that low. I suppose an alternative is to also provide research speed upgrades / have a way to give the ebay chrono boost somehow.

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml [hunk 2]
```diff
@@ -6518,6 +6518,34 @@
...
+                <Flags index="ToSelection" value="1"/>
+            </Button>
+        </InfoArray>
+        <InfoArray index="Research10" Time="220" Upgrade="AP_ZergFlyerArmorsLevel5">
+            <Resource index="Minerals" value="450"/>
+            <Resource index="Vespene" value="450"/>
+            <Button DefaultButtonFace="AP_zergflyerarmor5" State="Restricted" Requirements="AP_LearnZergFlyerArmor5">
+                <Flags index="ToSelection" value="1"/>
+            </Button>
+        </InfoArray>
```
Comment (MatthewMarinets) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml:6548]: Looks like a copy-paste error / adjustment necessary on the times here. Flyer weapon research time increases to L4 250 and L5 285, but armour stays at L3 220, L4 220, L5 220.

I'm in favour of just capping all L4/L5 research times to be the same as L3 research times, the issue here is just that it's inconsistent.

Comment (Ziktofel) [reply Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml:6548]: Fixed the typos

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml (hunk 1 above)
Comment (Ziktofel) [reply Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml:2346]: Done by reducing all W/A research times if L4 or L5 is available so the total research time is capped at 570

**Pull request merged**